Sources/SwiftGen/Array+Parallel.swift

```diff
@@ -0,0 +1,29 @@
+//
+// SwiftGen
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+
+// Based on https://github.com/realm/SwiftLint/blob/0.39.2/Source/SwiftLintFramework/Extensions/Array+SwiftLint.swift#
+
+extension Array {
+  func parallelFlatMap<T>(transform: (Element) -> [T]) -> [T] {
+    parallelMap(transform: transform).flatMap { $0 }
+  }
+
+  func parallelCompactMap<T>(transform: (Element) -> T?) -> [T] {
+    parallelMap(transform: transform).compactMap { $0 }
+  }
+
+  func parallelMap<T>(transform: (Element) -> T) -> [T] {
+    var result = ContiguousArray<T?>(repeating: nil, count: count)
+    return result.withUnsafeMutableBufferPointer { buffer in
```

Comment:
nit: I'm curious as to why using an UMBP here as opposed to accessing the `ContiguousArray` directly? Is it for performance reasons? Or for thread-safety reasons and avoid risk of concurrent access to the `ContinuousArray`?
I'm guessing the latter, but would be nice to add a code comment

Reply:
yeah, I think it's the latter, but i'm not super sure tbh 😬